### DISCUSSION TITLE
Store scoping issue with `ForEach`

### DISCUSSION THREAD

**Discussion by ddanilyuk** [pointfreeco/swift-composable-architecture#2822] [General]
Hi! Found an issue (i think) with scoping stores for lists.

Here is an example of a memory graph after opening and closing a feature with a List 10 times. After all this opening and closing, we have 70 cached stores of non-present rows. However, we should not have any of it since this feature is not present.
<img width="1728" alt="Screenshot 2024-02-18 at 01 04 56" src="https://github.com/pointfreeco/swift-composable-architecture/assets/26975779/b784c55f-2123-4e3b-aa1c-8e88c9f141bc">
This could potentially become a performance issue in large-scale apps.

Do we need a cleanup mechanism for the stores when the presented view closes? 
Or maybe we have a possibility to remove caching when using lists?

Repo with issue: [[link]](https://github.com/ddanilyuk/ListScopingIssue)
TCA version: 1.8.0
Xcode: 15.2

**Comment by stephencelis** [comment]
@ddanilyuk I’ve been traveling so haven’t had a chance to look into this till now, so surprised to see it closed. Was it not an issue after all? Can you elaborate on closing things?

**Reply by ddanilyuk** [reply]
Hi @stephencelis! I have opened an issue instead of a discussion because there was no reply.

**Reply by ddanilyuk** [reply]
https://github.com/pointfreeco/swift-composable-architecture/issues/2846
